Add subtree numba-dppy

PokhodenkoSA opened the pull request:
Command:
1. Create: `git subtree add --prefix=numba-dppy  https://github.com/IntelPython/numba-dppy.git main`
2. Pull changes from numba-dppy: `git subtree pull --prefix=numba-dppy https://github.com/IntelPython/numba-dppy.git main`

It could help to make PRs which require synchronous changes to `numba` and `numba-dppy`.

This PR should be merged, not squashed.

Commit:
(numba/dppy) Merge branch 'devel' of github.intel.com:SAT/NUMBA-PVC into devel

DrTodd13 committed:
(numba/dppy) Reduction work-in-progress.  Passing reduction var init value to kernel.  Adding reduction local array to kernel. Adding gufunc_template code to reduce across the local array.

DrTodd13 committed:
(numba/dppy) Adding the max work group size as an item stored in the device env.

reazulhoque committed:
(numba/dppy) Cleanedup llvm-spirv and spirv-tools requirements.

Commit:
(numba/dppy) change name of OCL to DPPY.

Commit:
(numba/dppy) Add debug prints to opencllite.c

Commit:
(numba/dppy) Updates to pairwise_distance_ocl.py example.

reazulhoque committed:
(numba/dppy) Added conda_env_path to ensure using the right binaries

reazulhoque committed:
(numba/dppy) Fix Error Msg

reazulhoque committed:
(numba/dppy) Remove Conda prefix to ensure correct bin

reazulhoque committed:
(numba/dppy) Added helper functions

reazulhoque committed:
(numba/dppy) Added biolerplate code to get started with adding unit tests

reazulhoque committed:
(numba/dppy) Updated renamed function calls in vector_add.c

Commit:
(numba/dppy) Remove dead codes and typos.

Commit:
(numba/dppy) Merge branch 'devel' of github.intel.com:SAT/NUMBA-PVC into devel

Commit:
(numba/dppy) Minor formatting change and remove incorrect comment.

Commit:
(numba/dppy) Add dump functions to driver classes.

Commit:
(numba/dppy) Minor formatting changes.

Commit:
(numba/dppy) Rename dppy.jit to dppy.kernel

Commit:
(numba/dppy) Update the sum.py example to use dppy.kernel

Commit:
(numba/dppy) Add two variants of sum.py for feature testing.

Commit:
(numba/dppy) Rename AutoJitDPPyKernel to JitDPPyKernel.

Commit:
(numba/dppy) conver EOL from windows to unix.

Commit:
(numba/dppy) Remove commented out code.

Commit:
(numba/dppy) dump_do_kernel_info should always print the kernel function name.

Commit:
(numba/dppy) Trun off caching of compiled kernels for the time being.

Commit:
(numba/dppy) Add another version of vector addition.

reazulhoque committed:
(numba/dppy) Added attributes to decorator for access_type

reazulhoque committed:
(numba/dppy) First test to perform sum on both CPU and GPU device

reazulhoque committed:
(numba/dppy) Added example using access_type for input parameter

Commit:
(numba/dppy) Use same data layout string as clang.

reazulhoque committed:
(numba/dppy) Added support for basic datatypes like int, flot, double and boolean.
Added test for making sure the scalars are being passed correctly.

reazulhoque committed:
(numba/dppy) Added check to make sure users know we do not support keyword arguments

Commit:
(numba/dppy) Unit test case for testing math routines.

reazulhoque committed:
(numba/dppy) Remove unwanted lines

reazulhoque committed:
(numba/dppy) Added support for basic datatypes

reazulhoque committed:
(numba/dppy) Fixed converting all int to long

reazulhoque committed:
(numba/dppy) Break long line into 80 chars

reazulhoque committed:
(numba/dppy) add test to make sure accessors are working

Commit:
(numba/dppy) Merge pull request #20 from SAT/feature/numba_array

Feature/directly pass ndarrays arguments to dppy.kernel

Commit:
(numba/dppy) Explicit error message for error code -54.

Commit:
(numba/dppy) Remove meminfo argument.

Commit:
(numba/dppy) Fix error check macro.

Commit:
(numba/dppy) Add a 2D vector sum example.

Commit:
(numba/dppy) Merge branch 'devel' of github.intel.com:SAT/NUMBA-PVC into devel

Commit:
(numba/dppy) Changes to modify how OpenCL global and local sizes are passed to a dppy.kernel function.

Commit:
(numba/dppy) Merge branch 'devel' into two-dimensional-kernels

Commit:
(numba/dppy) Generate executables with .exe suffix. Add *.exe to gitignore.

Commit:
(numba/dppy) Change to unix EOL.

Commit:
(numba/dppy) Update examples...

reazulhoque committed:
(numba/dppy) Make sure we are treating single int for global_work_size

reazulhoque committed:
(numba/dppy) Add Black Scholes test

reazulhoque committed:
(numba/dppy) Added sum reduction test

reazulhoque committed:
(numba/dppy) Updated pairwise_distance example

reazulhoque committed:
(numba/dppy) Make debug build an option

reazulhoque committed:
(numba/dppy) Remove files added accidentally

reazulhoque committed:
(numba/dppy) Updated global_wirk_size and local_work_size

Commit:
(numba/dppy) Reformat long comment.

Commit:
(numba/dppy) Merge branch 'devel' into ms138-directives

reazulhoque committed:
(numba/dppy) Remove errors.py from DPPy

reazulhoque committed:
(numba/dppy) Remove errors.py

reazulhoque committed:
(numba/dppy) Remove errors.py from DPPy

reazulhoque committed:
(numba/dppy) Merge branch 'ms138-directives' of https://github.intel.com/SAT/NUMBA-PVC into ms138-directives

reazulhoque committed:
(numba/dppy) Add dppy test to NUMBA test suite

reazulhoque committed:
(numba/dppy) Remove enums.py from DPPy

reazulhoque committed:
(numba/dppy) Add option to dump intermediate IR files

reazulhoque committed:
(numba/dppy) Moved flag to numba level

reazulhoque committed:
(numba/dppy) Remove not needed comments

reazulhoque committed:
(numba/dppy) Make argument optional

reazulhoque committed:
(numba/dppy) Added HowTo and reduction in examples

reazulhoque committed:
(numba/dppy) Added DPPy pipeline

Commit:
(numba/dppy) Merge branch 'ms138-directives' into spir-codegen

Commit:
(numba/dppy) Refactoring for coding style.

Commit:
(numba/dppy) Refactoring for coding style.

reazulhoque committed:
(numba/dppy) Added test to measure time of 1d reductions in different configurations

reazulhoque committed:
(numba/dppy) Added reduction plus scaling test

reazulhoque committed:
(numba/dppy) Added sync before timing

reazulhoque committed:
(numba/dppy) Include time to get data back

DrTodd13 committed:
(numba/dppy) Changes for reductions including addrspaces for vars.

reazulhoque committed:
(numba/dppy) Merge remote-tracking branch 'old/anl_release_v01' into devel

reazulhoque committed:
(numba/dppy) Added installation inside setup.py

reazulhoque committed:
(numba/dppy) Added changes to build conda package for linux

reazulhoque committed:
(numba/dppy) Fixed getting lib so intall and develop both works

Commit:
(numba/dppy) Merge branch 'devel' into ms138-directives

Commit:
(numba/dppy) Merge branch 'ms138-directives' into spir-codegen

reazulhoque committed:
(numba/dppy) Added recursive reduction example

Commit:
(numba/dppy) Refactor code to add a DPPy-specific lowerinbg pass.

reazulhoque committed:
(numba/dppy) Move an optimization level down and get rid of some addresscast

reazulhoque committed:
(numba/dppy) Add function attribute to always inline in LLVM ir

reazulhoque committed:
(numba/dppy) Added flag to turn spirv-val ON/OFF

reazulhoque committed:
(numba/dppy) Change how flag values work

reazulhoque committed:
(numba/dppy) Remove extra print

reazulhoque committed:
(numba/dppy) Increase test precision

reazulhoque committed:
(numba/dppy) Update examples

Commit:
(numba/dppy) Adding a basic square matrix multiply example.

reazulhoque committed:
(numba/dppy) Initial fix to get to the lowering pass using dppy_compiler_pass

reazulhoque committed:
(numba/dppy) Added dppy lowerer and dppy lowerer for arrayexpr

reazulhoque committed:
(numba/dppy) Inherit from Lower

reazulhoque committed:
(numba/dppy) Added DPPy target in njit

reazulhoque committed:
(numba/dppy) Added numpy_math_functions test and made numpy.add() work

reazulhoque committed:
(numba/dppy) Added the working math and trigonometric NumPy function tests

reazulhoque committed:
(numba/dppy) Replacing numpy ufuncs with OpenCL functions where possible and updated math and trigonometric tests

reazulhoque committed:
(numba/dppy) Addd test for numpy floating and bit-twiddling functions

reazulhoque committed:
(numba/dppy) Addd test for numpy comparison functions

Commit:
(numba/dppy) Remove unused code.

reazulhoque committed:
(numba/dppy) njit with parallel options will go through dppy pipeline

reazulhoque committed:
(numba/dppy) MAde sure parallel flag is only activated when provided or when target is dppy

reazulhoque committed:
(numba/dppy) Correct njit flag behavior

reazulhoque committed:
(numba/dppy) remove dppy dispatcher and create a new file cpu_dispatcher to avoid circular imports

reazulhoque committed:
(numba/dppy) Remove dppy backend (#44)

* Building NUMBA-DPPY successfully

* Removed some missed dppy_driver artifacts

* Removed new lines where it makes sense and replaced numba.dppy with .

reazulhoque committed:
(numba/dppy) Propagate CPUDispatcher being moved from targets/registry

reazulhoque committed:
(numba/dppy) Merge Devel

reazulhoque committed:
(numba/dppy) Initial cleaning of dppy_lowering

reazulhoque committed:
(numba/dppy) Clean up of call_dppy()

Commit:
(numba/dppy) Merge branch 'ms138-directives' into ms138-refactoring

Commit:
(numba/dppy) Remove oll dppy_driver from numba.

Commit:
(numba/dppy) convert the file to unix encoding.

Commit:
(numba/dppy) Convert to Unix encoding.

Commit:
(numba/dppy) Formatting changes...

Commit:
(numba/dppy) Refactor the code.

reazulhoque committed:
(numba/dppy) Some vectorize inliner is working

Commit:
(numba/dppy) More formatting changes...

Commit:
(numba/dppy) Run dos2unix on dppy_lowerer_2.py. Otherwise Eclipse on Windows does not recognize this as a Unix file.

Commit:
(numba/dppy) Merge branch 'ms138-directives' into ms138-refactoring

Commit:
(numba/dppy) Move DUFunc inliner into a new file and fix a bug. Blacksholes now works.

reazulhoque committed:
(numba/dppy) Remove pass that is now called dufunc_inliner

reazulhoque committed:
(numba/dppy) Added vectorize test

Commit:
(numba/dppy) Minor code style and function renaming.

Commit:
(numba/dppy) Merge branch 'ms138-directives' of github.intel.com:SAT/NUMBA-PVC into ms138-directives

reazulhoque committed:
(numba/dppy) Remove limit on numpy dimensions with simple loop scheduling

reazulhoque committed:
(numba/dppy) Fix indent calculation and 5d array example

reazulhoque committed:
(numba/dppy) remove variable for constant

Commit:
(numba/dppy) Move code hoisting and dppy function call generation to separate files.

Commit:
(numba/dppy) Remove suferfluous prints.

Commit:
(numba/dppy) Minor formatting changes.

Commit:
(numba/dppy) Rename dppy_lower_2 to dppy_lowerer

Commit:
(numba/dppy) add missing code from the original dppy_lowerer.py.

reazulhoque committed:
(numba/dppy) Support for atomic add (#46)

* LLVM ir is correctly generated, but llvm-spirv can not yet translate the llvm atomics to spirv

* Added atomic test

* Add atomic sub

* Fix sum example

* Remove unnecessary function

reazulhoque committed:
(numba/dppy) Ms138 directives (#52)

* Fix bugs to make pranges work

* Update README

* Update README

* Only keep setitems to track buffers being modified inside kernel

reazulhoque committed:
(numba/dppy) DPPY tests are passing

reazulhoque committed:
(numba/dppy) Feature/dppy func (#67)

* Fix bugs to make pranges work

* Update README

* Update README

* Only keep setitems to track buffers being modified inside kernel

* Added dppy.func that can be called from dppy.kernel

* Added reuse of dppy.func to test call from different dppy.kernels

* Fix line width

reazulhoque committed:
(numba/dppy) revert changes in dufunc to allow tests to pass

reazulhoque committed:
(numba/dppy) DPPy target will retain a copy of ufunc_db for replacing the implementation of numpy ufuncs with that of OpenCL's when such function exists

reazulhoque committed:
(numba/dppy) Added print to dppy.kernel (#69)

Commit:
(numba/dppy) Fix encoding of file from dos to unix. (#70)

Commit:
(numba/dppy) Add caching (#60)

* Add caching

* Del comment about errors in caching

* Add test for caching kernel

* Remove old commented out code.

reazulhoque committed:
(numba/dppy) Adds support for local memory allocation inside dppy.kernel, clean ups to barrier, use context manager

* Added local memory allocation and fixed bugs in barrier implementations

* Added support for passing signature in dppy.kernel decorator

* Added bigger example that deals with local memory and barrier

* use barrier and local memory in atomic test

* Change local.alloc to local.static_alloc to indicate this is static allocation

* Added pass to ensure only constants can be given as shape to static local memory allocation

* Change message and fix typo

* Include pass

* Feature/dppy backend context manager (#85)

* Do not use gpu_env instead use the current_env specified in the runtime.

* Update all examples to use dppy context manager.

* Fix Test suites and update API to use dppy context manager

* Uncomment barrier

* Fix caching tests

Commit:
(numba/dppy) Fix generation of the gufunc function name for parfors to make sure Intel Advisor works (#81)

* Create deterministic gufunc_name for njit kernel with parfor

PokhodenkoSA committed:
(numba/dppy) Migration to Numba 0.49 (#57)

* Unamed BasicBlock name can be just a number

* Force LLVM to use ZMM registers for vectorization.

https://reviews.llvm.org/D67259

* Fix usage of CPUDispatcher in TestWrapper tests.

Use CPUDispatcher from cpu_dispatcher module instead of registry module in numba.core package.

* Fix test_laziness (numba.tests.test_import.TestNumbaImport)

* Fix test_inlining (unittest.loader._FailedTest)

reazulhoque committed:
(numba/dppy) Merge branch 'devel' into 49.1

reazulhoque committed:
(numba/dppy) Update files with changes required to make them compatible with NUMBA 49.1

reazulhoque committed:
(numba/dppy) Updated to use sycllite branch from dppy-backend

reazulhoque committed:
(numba/dppy) Rename to dppy (#86)

PokhodenkoSA committed:
(numba/dppy) Force LLVM to use ZMM registers for vectorization.

https://reviews.llvm.org/D67259

reazulhoque committed:
(numba/dppy) Merge branch '49.1-gh' into '49.1'

49.1 gh

See merge request mlp-python/NUMBA-DPPY!74

reazulhoque committed:
(numba/dppy) Merge branch '49.1' into 'devel'

Rebase DPPY work on top of NUMBA-0.49.1

See merge request mlp-python/NUMBA-DPPY!54

reazulhoque committed:
(numba/dppy) Merged devel

reazulhoque committed:
(numba/dppy) Import ufunc_db only after lazy_init

reazulhoque committed:
(numba/dppy) Added check to not error out if OpenCL is absent

reazulhoque committed:
(numba/dppy) Fix minor bug

reazulhoque committed:
(numba/dppy) Rename njit option spirv to offload

PokhodenkoSA committed:
(numba/dppl) Rename DPPy to DPPL in code.

In some cases it is better to use Dppl as prefix for class names.

reazulhoque committed:
(numba/dppl) Added basic docstrings

reazulhoque committed:
(numba/dppl) Update examples and more docstrings

PokhodenkoSA committed:
(numba/dppl) Update to PyDPPL 0.1.1

reazulhoque committed:
(numba/dppl) Added unit test to check dppl apis

Commit:
(numba/dppl) Merge branch 'rhoque/update/dppl_api_unit_test' into 'devel'

Added unit test to check dppl apis

See merge request mlp-python/Numba-PyDPPL!82

reazulhoque committed:
(numba/dppl) Added atomics for sub and add

reazulhoque committed:
(numba/dppl) Add linkd spirv binary

reazulhoque committed:
(numba/dppl) Initial integration with numba.dppl.atomic.add

reazulhoque committed:
(numba/dppl) Finalize atomic add and sub for 4 datatypes

reazulhoque committed:
(numba/dppl) Made linking with external binaries optional

reazulhoque committed:
(numba/dppl) Atomic functions will be provided by pydppl

reazulhoque committed:
(numba/dppl) Update API and add test for local memory of different datatypes

reazulhoque committed:
(numba/dppl) Added atomic OpenCL kernel that will provide the atomic SPIR-V module

reazulhoque committed:
(numba/dppl) remove comment

PokhodenkoSA committed:
(numba/dppl) Use ctypes.util.find_library for finding DPPLOpenCLInterface and OpenCL (#7)

reazulhoque committed:
(numba/dppl) Initial try at calling a c++ function while lowering a function

AlexanderKalistratov committed:
(numba/dppl) Initial fallback implementation

reazulhoque committed:
(numba/dppl) Calling a function using address of it

reazulhoque committed:
(numba/dppl) Initial integration for calling a Numpy for gpu function

reazulhoque committed:
(numba/dppl) Add comments

reazulhoque committed:
(numba/dppl) Added skeleton for argsort, argmax,sum and dotvv

reazulhoque committed:
(numba/dppl) Add WIP cython layer to interface with dpnp

reazulhoque committed:
(numba/dppl) Retrieving the pointer correctly in cython

PokhodenkoSA committed:
(numba/dppl) Rename PyDPPL/dppl to dpCtl/dpctl (#49)

* Rename import from dppl to dpctl

* Rename in meta.yaml pydppl to dpctl

* Rename dependant package in readme from pydppl to dpCtl

* Rename backend in readme from PyDPPL to dpCtl

reazulhoque committed:
(numba/dppl) First stab

reazulhoque committed:
(numba/dppl) Added file back

reazulhoque committed:
(numba/dppl) merge with pydppl

reazulhoque committed:
(numba/dppl) Updated cov and dot

reazulhoque committed:
(numba/dppl) WIP added scalar args and submit

1e-to committed:
(numba/dppl) Added separate dispatcher (#42)

reazulhoque committed:
(numba/dppl) Added passing scalar to sycl backend

reazulhoque committed:
(numba/dppl) Sending reverse of global and local size to address semantic difference of sycl and opencl range

1e-to committed:
(numba/dppl) Added parfor lower message (#43)

reazulhoque committed:
(numba/dppl) Updated all examples and tests to use the new sycl backend WIP

reazulhoque committed:
(numba/dppl) Added exlicit usm creation

reazulhoque committed:
(numba/dppl) WIP

reazulhoque committed:
(numba/dppl) Added llvm generation for new backend

reazulhoque committed:
(numba/dppl) Merge branch 'pydppl' of https://github.com/reazulhoque/numba into feature/sycl_integration

1e-to committed:
(numba/dppl) Semantics "with context" (#57)

reazulhoque committed:
(numba/dppl) Fixed incorrect LLVM IR generation

reazulhoque committed:
(numba/dppl) Merge branch 'pydppl' into feature/sycl_integration

reazulhoque committed:
(numba/dppl) WIP

reazulhoque committed:
(numba/dppl) LLVM generation working

reazulhoque committed:
(numba/dppl) Cleanup

reazulhoque committed:
(numba/dppl) File name modification

reazulhoque committed:
(numba/dppl) Freeing the usm buffer being allocated

reazulhoque committed:
(numba/dppl) Merged pydppl

reazulhoque committed:
(numba/dppl) Merge branch 'feature/sycl_integration' into feature/experimental_numpy_for_gpu

reazulhoque committed:
(numba/dppl) successfully calling dpnp

reazulhoque committed:
(numba/dppl) Added more np functions

reazulhoque committed:
(numba/dppl) Move addition to typing templates of existing Numpy function into its own pass

reazulhoque committed:
(numba/dppl) Add check in setting lower extension

reazulhoque committed:
(numba/dppl) Added multidimensional test

reazulhoque committed:
(numba/dppl) Add bool

reazulhoque committed:
(numba/dppl) Added unload pass to undo the overloading

Commit:
(numba/dppl) Fix context

reazulhoque committed:
(numba/dppl) Fix for dpctl update

reazulhoque committed:
(numba/dppl) Merge branch 'feature/sycl_integration' into feature/experimental_numpy_for_gpu

reazulhoque committed:
(numba/dppl) Merge with feature/experimental_numpy_for_gpu

reazulhoque committed:
(numba/dppl) small update to documentation

reazulhoque committed:
(numba/dppl) Merge branch 'feature/sycl_integration' into feature/experimental_numpy_for_gpu

reazulhoque committed:
(numba/dppl) Merge branch 'feature/experimental_numpy_for_gpu' into feature/opencl_atomics

Commit:
(numba/dppl) Turn off debug mode after test

reazulhoque committed:
(numba/dppl) Fix getting dpctl lib at runtime

reazulhoque committed:
(numba/dppl) Merge branch 'pydppl' into feature/sycl_integration

reazulhoque committed:
(numba/dppl) Merge branch 'pydppl' into feature/experimental_numpy_for_gpu

reazulhoque committed:
(numba/dppl) Skip if dpnp is not present

reazulhoque committed:
(numba/dppl) Merge branch 'feature/experimental_numpy_for_gpu' into feature/opencl_atomics

reazulhoque committed:
(numba/dppl) skip tests if atomic support is not present

AlexanderKalistratov committed:
(numba/dppl) Add check if dpctl is present in decorator.
Add get_compiled to dispatcher.Dicpatcher so it behaves similar to TargetDispatcher

reazulhoque committed:
(numba/dppl) Turn lower_extension into a stack for parfor

reazulhoque committed:
(numba/dppl) Merge pull request #66 from AlexanderKalistratov/akalistr/fix_import_numba_without_dpctl

Add check if dpctl is present in decorator

AlexanderKalistratov committed:
(numba/dppl) Fix ir copy in fallback (#61)

AlexanderKalistratov committed:
(numba/dppl) Fix dppl_with_context example

AlexanderKalistratov committed:
(numba/dppl) Copy ir & funcdesc using same cache. Introduce new function to copy IR. Do not copy non-copyable objects (#78)

PokhodenkoSA committed:
Initial commit

PokhodenkoSA committed:
Update readme

This README.md is created having in mind the README for
https://github.com/numba/numba-scipy.

PokhodenkoSA committed:
Add license BSD-2-Clause

PokhodenkoSA committed:
Merge pull request #1 from IntelPython/spokhode/readme

Add README and License

PokhodenkoSA committed:
Add 'oneapi/' from commit 'c5b19175dfdef645dc7478b6e470340abb11201a'

git-subtree-dir: oneapi
git-subtree-mainline: 5c0ca1b4fe7729db6e96f6ac2e520e6a1cefab99
git-subtree-split: c5b19175dfdef645dc7478b6e470340abb11201a

PokhodenkoSA committed:
Add 'dppy/' from commit 'ad7d95cffb7513717b2b96545475f16d394ddb21'

git-subtree-dir: dppy
git-subtree-mainline: d5ddf3d79c77b6569c2a0b44a0718fcbe1f96820
git-subtree-split: ad7d95cffb7513717b2b96545475f16d394ddb21

PokhodenkoSA committed:
Add 'dppl/' from commit 'fa996e3adbfef446d97c4f3b7f6565f57abc9e40'

git-subtree-dir: dppl
git-subtree-mainline: c14404586ae21a5cfacd42281d5b4c627187bb28
git-subtree-split: fa996e3adbfef446d97c4f3b7f6565f57abc9e40

PokhodenkoSA committed:
Add 'numba_dppy/' from commit 'fa996e3adbfef446d97c4f3b7f6565f57abc9e40'

git-subtree-dir: numba_dppy
git-subtree-mainline: 57b68906898100a378e9eea6ef0ce57db5f42e16
git-subtree-split: fa996e3adbfef446d97c4f3b7f6565f57abc9e40

PokhodenkoSA committed:
Remove history folders: oneapi, dppy, dppl

History folders contains history for numba/oneapi, numba/dppy and
numba/dppl folders.

PokhodenkoSA committed:
Merge pull request #2 from IntelPython/spokhode/add-dppy

Move `numba/dpp`l to `numba_dppy` with all history

diptorupd commented on numba-dppy/LICENSE:
```diff
@@ -0,0 +1,24 @@
+Copyright (c) 2012, Anaconda, Inc.
```
Should this not be our License file?

PokhodenkoSA replied:
This is license file from numba-dppy. I think it contains an error.

PokhodenkoSA replied:
Updated.

diptorupd commented on numba-dppy/README.md:
```diff
@@ -0,0 +1,21 @@
+# numba-dppy
+
+## Numba + dpCtl + dpNP = numba-dppy
```
maybe Numba + DPPY + dpCtl + dpNP = numba-dppy :)

PokhodenkoSA replied:
@diptorupd please, provide a PR to https://github.com/IntelPython/numba-dppy.git

PokhodenkoSA committed:
Update LICENSE (#3)

PokhodenkoSA committed:
Remove context_name (#6)

See https://github.com/IntelPython/numba/pull/100#issuecomment-726890717

PokhodenkoSA committed:
Add 'numba-dppy/' from commit '7b1d06b763bdf773a8191a6660cf5343dd5c4d39'

git-subtree-dir: numba-dppy
git-subtree-mainline: 943884ff997a19a9a773205e18873c435c60020a
git-subtree-split: 7b1d06b763bdf773a8191a6660cf5343dd5c4d39
